metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/dsl/functions/MapFunctions.java

```diff
@@ -82,4 +84,71 @@ public Object apply(List<Object> objects) {
       return ret;
     }
   }
+
+  @Stellar(name = "PUT",
+           namespace = "MAP",
+           description = "Adds a key/value pair to a map",
+           params = {
+                      "key - The key",
+                      "value - The value",
+                      "map - The map to perform the put on"
+                    },
+           returns = "The original map modified with the key/value. If the map argument is null, a new map will be created and returned that contains the provided key and value - note: if the 'map' argument is null, only the returned map will be non-null and contain the key/value."
+          )
+  public static class MapPut extends BaseStellarFunction {
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Object apply(List<Object> objects) {
+      if (objects.size() < 3) {
+        throw new IllegalArgumentException("Must pass a key, value, and map");
+      } else {
```

Comment:
Do we not want type checking at all here?  This is a pretty specific function

Reply:
What kind of type checking do you think we should add? The map should accept any types for keys and values, and the map argument itself will throw a ClassCastException if it's not actually a map. I tried to make the functionality pretty consistent with our examples, e.g. SET_ADD https://github.com/mmiklavc/metron/blob/a8628f1610a40cd0c790eb0b3caf6d2585db3018/metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/dsl/functions/SetFunctions.java#L64

Reply:
If you search for `instanceof` in stellar common you will see that we do handle it, if only to return null at times.  In many places.
We are certainly inconsistent on this point.  My preference would be to handle it, since we do not have explicitly typed parameters or parameter checking